block/manager.go

```diff
@@ -606,8 +606,8 @@ func (m *Manager) publishBlockInternal(ctx context.Context) error {
 	header.Signature = signature
 
 	if err := header.ValidateBasic(); err != nil {
-		// TODO(tzdybal): I think this is could be even a panic, because if this happens, header is FUBAR
-		m.logger.Error("header validation error", "error", err)
+		// If this ever happens, for reovery, check for a mismatch between the configured signing key and the proposer address in the genesis file
```

Comment:
it seems like we are applying the block on the execution environment here. this panic looks like it could leave the application in an unusable state. In the sdk if setfinal is not called then we can get around not writing to disk but does this hold for reth and evm?

Reply:
Moved this validation logic before `applyBlock` so this is not the case anymore. setFinal corresponds to the `engine_forkChoiceUpdatedV3` method: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#engine_forkchoiceupdatedv3